common/chains/client/rpc_client.go

```diff
@@ -0,0 +1,83 @@
+package client
+
+import (
+	"context"
+	"math/big"
+
+	feetypes "github.com/smartcontractkit/chainlink/v2/common/fee/types"
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+	"github.com/smartcontractkit/chainlink/v2/core/assets"
+)
+
+type RPCClient[
+	CHAINID types.ID,
+	SEQ types.Sequence,
+	ADDR types.Hashable,
+	BLOCK any,
+	BLOCKHASH types.Hashable,
+	TX any,
+	TXHASH types.Hashable,
+	EVENT any,
+	EVENTOPS any, // event filter query options
+	TXRECEIPT any,
+	FEE feetypes.Fee,
+	HEAD types.Head[BLOCKHASH],
+	SUB types.Subscription,
+] interface {
+	Accounts[ADDR, SEQ]
+	Transactions[TX, TXHASH, TXRECEIPT]
+	Events[EVENT, EVENTOPS]
+	Blocks[BLOCK, BLOCKHASH]
+
+	BatchCallContext(ctx context.Context, b []any) error
+	CallContract(
+		ctx context.Context,
+		attempt interface{},
+		blockNumber *big.Int,
+	) (rpcErr []byte, extractErr error)
+	CallContext(ctx context.Context, result interface{}, method string, args ...interface{}) error
+	ChainID() (CHAINID, error)
+	CodeAt(ctx context.Context, account ADDR, blockNumber *big.Int) ([]byte, error)
+	ConfiguredChainID() CHAINID
+	EstimateGas(ctx context.Context, call any) (gas uint64, err error)
+	HeadByNumber(ctx context.Context, number *big.Int) (head HEAD, err error)
+	HeadByHash(ctx context.Context, hash BLOCKHASH) (head HEAD, err error)
+	LINKBalance(ctx context.Context, accountAddress ADDR, linkAddress ADDR) (*assets.Link, error)
+	PendingSequenceAt(ctx context.Context, addr ADDR) (SEQ, error)
+	SendEmptyTransaction(
```

Comment:
better place is Transactions interface?

Reply:
I'm assuming you meant just SendEmptyTransaction to the Accounts interface, done!

Reply:
`SendEmptyTransaction` is the only method that uses FEE. Is there a way we can restructure this method so we can avoid using FEE? It is propagated everywhere, so it would be nice to not have it.

Reply:
Where it's used in the txmgr client seems to need FEE to create a new TX attempt. https://github.com/smartcontractkit/chainlink/blob/516bd394d6ab72564af51aeeebf7ad48b8317b11/core/chains/evm/txmgr/client.go#L139-L163. It isn't used by the EVM client though, so possibly we could get rid of the function altogether?

Reply:
This can be skipped now, since the API was simplified.